src/operations/generate.rs

```diff
@@ -45,20 +45,16 @@ fn color_pass(
     min_saturation: Option<f32>,
     max_saturation: Option<f32>,
 ) -> Option<Rgb> {
-    let mut chosen = None;
-    for color in colors {
-        let (saturation, luma) = grab_sat_luma(*color);
-        if (max_luma == None || luma <= max_luma.unwrap())
+    let predicate = |rgb: &Rgb| {
```

Comment:
I wish this closure had a better name but to be honest I have very little clue on what it does